File: JAVA/Program08.java
////////////////////////////////////////////////////////////////////////////////////////     
//                                                                                    //
//  File name :     Program08.java                                                    //
//  Description :   Used to Display Total Count of Non-Factors of Given Number        //
//  Author :        Pallavi Omprakash Malewar                                         //
//  Date :          24/08/2025                                                        //
//                                                                                    //
////////////////////////////////////////////////////////////////////////////////////////

import java.util.Scanner;

class Number
{
    public int totalNonFactors(int iNo)
    {
        int iCount = 0, iCnt = 0;

        for(iCount = 1 ; iCount <= iNo ; iCount++)
        {
            if((iNo % iCount) != 0)
            {
                iCnt++;         // To Count Non Factors
            }
        }
        return iCnt;
    }
}


class Program08
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        int iValue = 0, iRet = 0 ;

        System.out.print("Enter Number : ");
        iValue = sobj.nextInt();

        Number nobj = new Number();

        iRet = nobj.totalNonFactors(iValue);

        System.out.println("Total Count of Non-Factors of " + iValue + " is "+ iRet);

        sobj.close();

    }
}

File: CPP/Program08.cpp
/////////////////////////////////////////////////////////////////////////////////////////////////////
//                                                                                                 //
//  File name :     Program08.cpp                                                                  //
//  Description :   Used to take number and display the Count of one in the binary converion       //
//  Author :        Pallavi Omprakash Malewar                                                      //
//  Date :          05/09/2025                                                                     //
//                                                                                                 //
/////////////////////////////////////////////////////////////////////////////////////////////////////


#include<iostream>
using namespace std;

int CountOne(int iNo)
{
    int iDigit = 0;
    int iCount = 0;

    while(iNo != 0)
    {
        iDigit = iNo % 2;
        if(iDigit == 1)
        {
            iCount++;
        }
        iNo = iNo / 2;
    }
    return iCount;
}


int main()
{
    int iValue = 0, iRet = 0;

    cout<<"Enter Number : "<<endl;
    cin>>iValue;

    iRet = CountOne(iValue);

    cout<<"Count of 1 : "<<iRet<<endl;

    return 0;
}